iothub/service/src/Common/Exceptions/ServerErrorException.cs

```diff
@@ -7,8 +7,11 @@
 namespace Microsoft.Azure.Devices.Common.Exceptions
 {
     /// <summary>
-    /// The exception that is thrown when IoT Hub encounters an error while processing a request.
+    /// The exception that is thrown when the IoT Hub returned an internal service error.
     /// </summary>
+    /// <remarks>
+    /// This exception typically means the IoT Hub service has encountered an unexpected error and is usually transient. Please review the <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-troubleshoot-error-500xxx-internal-errors">500xxx Internal errors</see> guide for more information. The best course of action is to retry your operation after some time. By default, the SDK will utilize the <see cref="ExponentialBackoff"/> retry strategy.
+    /// </remarks>
```

Comment:
Can point to - https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/devdoc/retrypolicy.md